### PR TITLE
Flask hw5

### PR DESCRIPTION
**Pull request by OleksUMD**

### CODE REVIEW

File: flask_hw5/blueprints/rooms/rooms_routes.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+from hotel_db import db
+
+
+class Rooms(Resource):
+    @marshal_with_decor
+    def get(self):
+        room_filters = room_parser.parse_args()
+        if any(tuple(room_filters.values())):
+            return Room.query.filter_by(
+                                        **{k: v
```
Comment (dimigor) [inline flask_hw5/blueprints/rooms/rooms_routes.py]: ohhh, bad formating

File: flask_hw5/blueprints/rooms/rooms_routes.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                                           for k, v in room_filters.items()
+                                           if v}
+                                       ).first()
+        return Room.query.all(), 200
+
+    def post(self):
+        new_room_values = room_parser_post.parse_args()
+        db.session.add(Room(**new_room_values))
+        db.session.commit()
+        return 'New room was successfully added!', 200
```
Comment (dimigor) [inline flask_hw5/blueprints/rooms/rooms_routes.py]: return added room. Client should know what fields you added

File: flask_hw5/blueprints/rooms/rooms_routes.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                                        **{k: v
+                                           for k, v in room_filters.items()
+                                           if v}
+                                       ).first()
+        return Room.query.all(), 200
+
+    def post(self):
+        new_room_values = room_parser_post.parse_args()
+        db.session.add(Room(**new_room_values))
+        db.session.commit()
```
Comment (dimigor) [inline flask_hw5/blueprints/rooms/rooms_routes.py]: what if something went wrong during adding new room? you should handle this cases on every methods

**Review by dimigor — CHANGES_REQUESTED** [review]

File: flask_hw5/blueprints/rooms/rooms_routes.py (hunk 1 above)
Comment (OleksUMD) [reply flask_hw5/blueprints/rooms/rooms_routes.py]: A little bit fixed

File: flask_hw5/blueprints/rooms/rooms_routes.py (hunk 2 above)
Comment (OleksUMD) [reply flask_hw5/blueprints/rooms/rooms_routes.py]: Done

File: flask_hw5/blueprints/rooms/rooms_routes.py (hunk 3 above)
Comment (OleksUMD) [reply flask_hw5/blueprints/rooms/rooms_routes.py]: Done!

File: flask_hw5/blueprints/rooms/rooms_routes.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        if any(tuple(room_filters.values())):
+            return Room.query.filter_by(**{k: v
+                                           for k, v in room_filters.items()
+                                           if v}).first_or_404()
+        return Room.query.all(), 200
+
+    @marshal_with_decor
+    def post(self):
+        new_room_values = room_parser_post.parse_args()
+        room_to_add = Room(**new_room_values)
```
Comment (dimigor) [inline flask_hw5/blueprints/rooms/rooms_routes.py:25]: an exception may be here if you pass unknown keys? or do not pass any of required keys?

File: flask_hw5/blueprints/rooms/rooms_routes.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        db.session.add(room_to_add)
+        try:
+            db.session.commit()
+        except SQLAlchemyError:
+            return 'Something wrong happened on our side. Please, try later!'
+        return room_to_add, 200
```
Comment (dimigor) [inline flask_hw5/blueprints/rooms/rooms_routes.py]: return `201` code -> created

File: flask_hw5/blueprints/rooms/rooms_routes.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+    def patch(self, number):
+        room_info_to_update = room_parser_patch.parse_args()
+        room_to_update = Room.query.filter_by(number=number).first_or_404()
+        if not room_to_update:
+            return 'Sorry the room you are looking for was not found.'
```
Comment (dimigor) [inline flask_hw5/blueprints/rooms/rooms_routes.py]: code 400 or 404

File: flask_hw5/blueprints/staff/staff_routes.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+        return StaffObj.query.all()
+
+    def post(self):
+        new_staff_values = staff_parser_post.parse_args()
+        db.session.add(StaffObj(**new_staff_values))
+        try:
+            db.session.commit()
+        except SQLAlchemyError:
+            return 'Something wrong happened on our side. Please, try later!'
+        return 'New staff was successfully added!'
```
Comment (dimigor) [inline flask_hw5/blueprints/staff/staff_routes.py]: as above, return data

**Review by dimigor — CHANGES_REQUESTED** [review]

File: flask_hw5/blueprints/staff/staff_routes.py (hunk 7 above)
Comment (OleksUMD) [reply flask_hw5/blueprints/staff/staff_routes.py]: Done

File: flask_hw5/blueprints/rooms/rooms_routes.py (hunk 6 above)
Comment (OleksUMD) [reply flask_hw5/blueprints/rooms/rooms_routes.py]: Updated. if else in this case is not required. first_or_404 method manages this problem.

File: flask_hw5/blueprints/rooms/rooms_routes.py (hunk 5 above)
Comment (OleksUMD) [reply flask_hw5/blueprints/rooms/rooms_routes.py]: Done

File: flask_hw5/blueprints/rooms/rooms_routes.py (hunk 4 above)
Comment (OleksUMD) [reply flask_hw5/blueprints/rooms/rooms_routes.py:25]: custom parser catch all of those. No exceptions occurred in cases like: empty fields, wrong fields, additional unknown fields, fields with wrong data types etc. Of course, we can implement more precise validation with flask_form and wtf_validators(I am going to do it in my flask project), but I am not sure it is needed in  tasks like this one.

File: flask_hw5/blueprints/rooms/rooms_routes.py [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    def patch(self, number):
+        room_info_to_update = room_parser_patch.parse_args()
+        room_to_update = Room.query.filter_by(number=number).first_or_404()
+        for key, value in room_info_to_update.items():
+            if value:
+                setattr(room_to_update, key, value)
+        try:
+            db.session.commit()
+        except SQLAlchemyError:
+            return 'Something wrong happened on our side. Please, try later!'
```
Comment (dimigor) [inline flask_hw5/blueprints/rooms/rooms_routes.py:42]: also need to specify http status code

**Review by dimigor — APPROVED** [review]

**Pull request merged**